src/input/mod.rs

```diff
@@ -849,15 +863,22 @@ impl State {
                                             if let Some(surface) =
                                                 target.toplevel().map(Cow::into_owned)
                                             {
+                                                // If the logo is held then the pointer event is
+                                                // aimed at the compositor and shouldn't be passed
+                                                // to the application.
+                                                pass_event = false;
```

Comment:
We should add the button to supressed_buttons right once we decided to not pass the event. Otherwise we might get mismatches, as it isn't deterministic when the idle_callbacks will run. Also having this in the `_`-match clause and `dispatch_grab` seems highly repetitive. We know already that we will skip the event here.

Though I would like to chance even this logic. Currently we discard every mouse button, if super as held, but we only need to discard those, that we will handle ourselves. So we should not prevent middle-clicks and other buttons, just left and right clicks.

Reply:
Then we should also not need to add this to the grab code, which I really want to avoid as that is leaking abstractions left and right.

Reply:
> Then we should also not need to add this to the grab code, which I really want to avoid as that is leaking abstractions left and right.

The problem is that `ptr.button` needs to be called if there is a grab because otherwise it won't receive the release event and it will be stuck moving/resizing.

And at that point the grab is the one that forwards to the client so it must be the one responsible for suppressing the button.

Reply:
We could also release the grab inside of the input-handler. Its drop code will be called either way.